tests/main/snap-userd-socket-activation/task.yaml

```diff
@@ -0,0 +1,47 @@
+summary: Check that root can activate userd via socket activation
+
+systems:
+    - -ubuntu-core-*
+    - -ubuntu-14.04-*
+    - -amazon-linux-2-*
+    - -centos-7-*
+
+environment:
+    TEST_UID: $(id -u test)
+    USER_RUNTIME_DIR: /run/user/${TEST_UID}
+
+prepare: |
+    mkdir -p "$USER_RUNTIME_DIR"
+    chmod u=rwX,go= "$USER_RUNTIME_DIR"
+    chown test:test "$USER_RUNTIME_DIR"
+    systemctl start "user@${TEST_UID}.service"
```

Comment:
What is this service? Is that the generic systemd-for-user-session?

Reply:
It is a system level unit that will start a user mode systemd instance for the given uid.  While with old versions, it was sufficient to just run `systemd --user` to start a user instance, that doesn't work with modern versions since it checks e.g. that the cgroup configuration is correct.